src/Files.App/ViewModels/Properties/SecurityViewModel.cs

```diff
@@ -195,63 +200,68 @@ private async void EditOwner()
 				}
 			}
 		}
 
-		private void RemoveRulesForUser()
+		private void RemoveACE()
 		{
-			if (SelectedRuleForUser is not null)
+			if (SelectedAccessControlEntry is not null)
 			{
-				SelectedRuleForUser.AllowRights = 0;
-				SelectedRuleForUser.DenyRights = 0;
+				SelectedAccessControlEntry.AllowedAccessMaskFlags = 0;
+				SelectedAccessControlEntry.DeniedAccessMaskFlags = 0;
 
-				if (!FilePermissions.AccessRules.Any(x => x.IdentityReference == SelectedRuleForUser.UserGroup.Sid))
+				if (!AccessControlListController.AccessControlEntriesAdvanced.Any(x => x.PrincipalSid == SelectedAccessControlEntry.Principal.Sid))
 				{
 					// No remaining rules, remove user from list
-					FilePermissions.RulesForUsers.Remove(SelectedRuleForUser);
+					AccessControlListController.AccessControlEntries.Remove(SelectedAccessControlEntry);
 				}
 			}
 		}
 
 		public void GetFilePermissions()
 		{
-			bool isFolder = Item.PrimaryItemAttribute == Windows.Storage.StorageItemTypes.Folder && !Item.IsShortcut;
-			FilePermissions = new FilePermissionsManager(FileOperationsHelpers.GetFilePermissions(Item.ItemPath, isFolder));
+			bool isFolder = Item.PrimaryItemAttribute == StorageItemTypes.Folder && !Item.IsShortcut;
+
+			var acl = FileOperationsHelpers.GetFilePermissions(Item.ItemPath, isFolder);
+			AccessControlListController = new(acl);
 		}
 
 		public bool SetFilePermissions()
 		{
-			if (FilePermissions is null || !FilePermissions.CanReadFilePermissions)
+			// If user has no permission to change ACL
+			if (AccessControlListController is null ||
+				!AccessControlListController.CanReadAccessControl)
 				return true;
 
-			return FilePermissions.ToFilePermissions().SetPermissions();
+			return AccessControlListController.ToAccessControlList().SetAccessControl();
 		}
 
 		public bool SetFileOwner(string ownerSid)
 		{
-			bool isFolder = Item.PrimaryItemAttribute == Windows.Storage.StorageItemTypes.Folder && !Item.IsShortcut;
+			bool isFolder = Item.PrimaryItemAttribute == StorageItemTypes.Folder && !Item.IsShortcut;
 
 			return FileOperationsHelpers.SetFileOwner(Item.ItemPath, isFolder, ownerSid);
 		}
 
-		public Task<string?> OpenObjectPicker()
-			=> FileOperationsHelpers.OpenObjectPickerAsync(NativeWinApiHelper.CoreWindowHandle.ToInt64());
-
-		public bool SetAccessRuleProtection(bool isProtected, bool preserveInheritance)
+		public bool SetAccessControlProtection(bool isProtected, bool preserveInheritance)
 		{
-			bool isFolder = Item.PrimaryItemAttribute == Windows.Storage.StorageItemTypes.Folder && !Item.IsShortcut;
+			bool isFolder = Item.PrimaryItemAttribute == StorageItemTypes.Folder && !Item.IsShortcut;
 
 			return FileOperationsHelpers.SetAccessRuleProtection(Item.ItemPath, isFolder, isProtected, preserveInheritance);
 		}
+
+		public static Task<string?> OpenObjectPicker()
+			=> FileOperationsHelpers.OpenObjectPickerAsync(NativeWinApiHelper.CoreWindowHandle.ToInt64());
```

Comment:
Can you remove arrow function?

Reply:
Why? Should be? I didn't change to use arrow, though.